src/processors/session-processor.ts

```diff
@@ -35,18 +35,25 @@ export default class SessionProcessor<T extends Session> extends KnexProcessor<T
       throw jsonApiErrors.AccessDenied();
     }
 
-    const userId = String(user.id);
+    const userPk = userType.schema.primaryKeyName || DEFAULT_PRIMARY_KEY;
+    const passwordFields = Object.keys(userType.schema.attributes).filter(
+      attribute => userType.schema.attributes[attribute] === Password
+    );
+    const userId = String(user[userPk]);
 
-    delete user.id;
+    [userPk, ...passwordFields].forEach(field => delete user[field]);
```

Comment:
Wouldn't it be better to destructure the user into the props you want?
Something like 
`const { id, ...passwordFields, ...cleanUser} = user;`
and then just use the cleanUser

Reply:
That looks better, actually. I'll refactor it that way.

Reply:
Resolved in #140.